static/app/views/eventsV2/results.tsx

```diff
@@ -103,7 +104,11 @@ function getYAxis(location: Location, eventView: EventView, savedQuery?: SavedQu
 
 export class Results extends Component<Props, State> {
   static getDerivedStateFromProps(nextProps: Readonly<Props>, prevState: State): State {
-    if (nextProps.savedQuery || !nextProps.loading) {
+    if (
+      ((!nextProps.isHomepage || prevState.savedQuery) &&
+        (nextProps.savedQuery || !nextProps.loading)) ||
+      nextProps.savedQuery === undefined // When user clicks on Discover in sidebar
```

Comment:
```suggestion
      !nextProps.isHomepage ||
      prevState.savedQuery ||
      nextProps.savedQuery === undefined // When user clicks on Discover in sidebar
```

I think this is equivalent? We don't need to check for `(nextProps.savedQuery || !nextProps.loading)` because we enter the if clause if `nextProps.savedQuery === undefined` anyways

Reply:
Done! :tada: